src/sync.js

```diff
@@ -0,0 +1,165 @@
+}
+
+function decrypt(key, encrypted) {
+  const decrypter = new JoseJWE.Decrypter(cryptographer, shared_key(key));
+  return decrypter.decrypt(encrypted).then(result => {
+    return JSON.parse(result);
+  });
+}
+
+
+function loadFromKinto(client) {
+  // Get credentials and lastmodified
+  browser.storage.local.get(['credentials', 'contentWasSynced', 'last_modified'])
+    .then((data) => {
+      // XXX: Ask for an refresh token
+      // Query Kinto with the Bearer Token
+      if (!data.hasOwnProperty('credentials')) return;
+
+      client
+        .bucket('default')
+        .collection('notes')
+        .getRecord('singleNote', {
+          headers: { Authorization: `Bearer ${data.credentials.access_token}` }
+        })
+        .then(result => {
+          if (!data.hasOwnProperty('last_modified') ||
+              result.data.last_modified > data.last_modified) {
+            // If there is something in Kinto send unencrypted content to the sidebar
+            return decrypt(data.credentials.key, result['data']['content'])
+              .then(content => {
+                browser.runtime.sendMessage({
+                  action: 'kinto-loaded',
+                  data: content,
+                  contentWasSynced: data.contentWasSynced,
+                  last_modified: result.data.last_modified
+                });
+              })
+              .catch(() => {
+                // In case we cannot decrypt the message
+                if (data.credentials.key.kid < result.data.kid) {
+                  // If the key date is greater than current one, log the user out.
+                  return browser.storage.local.remove('credentials');
+                } else {
+                  // If the key date is older than the current one, we can't help
+                  // because there is no way we get the previous key.
+                  // Flush the server ALA sync
+                  return client
+                    .bucket('default')
+                    .collection('notes')
+                    .deleteRecord('singleNote', {
+                      headers: { Authorization: `Bearer ${data.credentials.access_token}` }
+                    });
+                }
+              });
+          }
+        })
+        .catch(error => {
+          if (/HTTP 404/.test(error.message)) {
+            // If there is nothing in Kinto send null to the sidebar
+            console.log('Kinto is emtpy');
+            browser.runtime.sendMessage({
+              action: 'kinto-loaded',
+              data: null
+            });
+          } else if (/HTTP 401/.test(error.message)) {
+            // In case of 401 log the user out.
+            return browser.storage.local.remove('credentials');
+          } else {
+            console.error(error);
+          }
+        });
+    });
+}
+
+function saveToKinto(client) {
+  // XXX: Debounce the call and set the status to Editing
+  browser.runtime.sendMessage('notes@mozilla.com', {
```

Comment:
I don't pretend I understand the message sending stuff, but I'll _assume_ that we're OK with it sometimes being `chrome.runtime.sendMessage()` vs `browser.runtime.sendMessage()` and sometimes `browser.runtime.sendMessage('notes@mozilla.com', {...})` vs other times where we don't specify the `'notes@mozilla.com'` as the first arg.

Reply:
Yes we specify the ID of the addons when we want to send the event to all open sidebars rather than when we just want to send it to the current one.

The only benefit of using chrome instead of browser is if we want to support to run the add-on in Chrome, which is not the case since we added Firefox Account.